### PR TITLE
CNKAF-951: Add catchingCompose utility to handle exceptions.

### PR DESCRIPTION
**Pull request by rigelbm**
Usage:

```
catchingCompose(
   future,
   MyException.class,
   myException -> doSomethingWithMyException(myException));
```

### CODE REVIEW

File: kafka-rest/src/main/java/io/confluent/kafkarest/common/CompletableFutures.java [hunk 1]
```diff
@@ -46,4 +48,32 @@ private CompletableFutures() {
...
+    return future.handle(
+        (value, error) -> {
+          if (error == null) {
+            return CompletableFuture.completedFuture(value);
+          } else if (exceptionClass.isInstance(error.getCause())) {
+            return handler.apply(exceptionClass.cast(error.getCause()));
+          } else if (error instanceof CompletionException) {
+            throw (CompletionException) error;
+          } else {
+            throw new AssertionError(error); // impossible
```
Comment (rnpridgeon) [inline kafka-rest/src/main/java/io/confluent/kafkarest/common/CompletableFutures.java:74]: nit: replace with "Should not happen"; if it's truly impossible this else need not exist.

File: kafka-rest/src/main/java/io/confluent/kafkarest/common/CompletableFutures.java [hunk 2]
```diff
@@ -46,4 +48,32 @@ private CompletableFutures() {
...
     return future;
   }
+
+  /**
+   * Returns a new {@link CompletableFuture} that is completed when {@code future} is complete,
+   * catching the given {@link Throwable exceptionClass}.
+   *
+   * <p>If {@code future} completes normally, then the returned future completes normally with the
+   * same value. If {@code future} completes exceptionally with an {@code exceptionClass}, the
+   * returned completes with the result of the {@code handler}. Otherwise, the returned future will
```
Comment (rnpridgeon) [inline kafka-rest/src/main/java/io/confluent/kafkarest/common/CompletableFutures.java:58]: "the returned future"?

File: kafka-rest/src/main/java/io/confluent/kafkarest/common/CompletableFutures.java (hunk 1 above)
Comment (rigelbm) [reply kafka-rest/src/main/java/io/confluent/kafkarest/common/CompletableFutures.java:74]: Done.

File: kafka-rest/src/main/java/io/confluent/kafkarest/common/CompletableFutures.java (hunk 2 above)
Comment (rigelbm) [reply kafka-rest/src/main/java/io/confluent/kafkarest/common/CompletableFutures.java:58]: Done.

**Pull request merged**